python/rmm/_lib/device_buffer.pyx

```diff
@@ -354,9 +351,6 @@ cdef void _copy_async(const void* src,
     if err != cudaError.cudaSuccess:
         raise RuntimeError(f"Memcpy failed with error: {err}")
 
-    if stream.is_default():
-        stream.synchronize()
```

Comment:
We were relying on this to prevent race conditions with regards to mutable host buffers. We'd need to insert synchronization where this `_copy_async` function was being used and expects synchronization when copying host memory on the default stream.